src/sensors/LinearSensor.py

```diff
@@ -0,0 +1,49 @@
+from src.services.ADC import ADC
+from src.test_lib.input import Input
+from src.test_lib.condition import Condition
+import asyncio
+
+class LinearSensor():
+    def __init__(self,adc:ADC,slope:float, offset:float):
+        self._adc=adc
+        self.slope=slope
+        self.offset=offset
+        self.value=0.0
+    
+    def write(self,value:float):
+        self.value=value
+        adc_value=(self.value-self.offset)/self.slope
+        input_action=self._adc.generate_value(adc_value)
+        input_action.apply()
+
+    def generate_case(self, initial_value: float, final_value: float, delay: float):
```

Comment:
A generate_case should just take one value, and generate that value, if in the future we want to create sweeps, we will do so as we need it

Reply:
also the name could be something like `generate_reading`